File: domain/src/commonMain/kotlin/com/voxenlabs/domain/server/usecases/RemoveStoredServerUseCase.kt
package com.voxenlabs.domain.server.usecases

import com.voxenlabs.domain.server.models.Server
import com.voxenlabs.domain.server.repositories.ServerRepositoryInterface
import org.koin.core.annotation.Factory

@Factory
class RemoveStoredServerUseCase(
    val serverRepository: ServerRepositoryInterface,
) {
    suspend operator fun invoke(url: String) = serverRepository.removeStoredServer(url)
}

File: src/modules/Voxen.Client.Domain/ServerDefinitions/UseCases/RemoveStoredServerUseCase.cs
using Voxen.Client.Domain.ServerDefinitions.Models;
using Voxen.Client.Domain.ServerDefinitions.RepositoryInterfaces;

namespace Voxen.Client.Domain.ServerDefinitions.UseCases;

public class RemoveStoredServerUseCase(IServerRepository serverRepository)
{
    private readonly IServerRepository serverRepository = serverRepository;

    public void Invoke(Server server)
    {
        serverRepository.RemoveStoredServer(server);
    }
}
